base/MAPL_ObsUtil.F90

```diff
@@ -668,6 +736,68 @@ subroutine sort_four_arrays_by_time(U,V,T,ID,rc)
   end subroutine sort_four_arrays_by_time
 
 
+  subroutine sort_index (X, IA, rc)
+    use MAPL_SortMod
+    real(ESMF_KIND_R8), intent(in) :: X(:)
+    integer, intent(out) :: IA(:)            ! index
+    integer, optional, intent(out) :: rc
+
+    integer :: i, len
+    integer(ESMF_KIND_I8), allocatable :: IX(:)
+
+    _ASSERT (size(X)==size(IA), 'X and IA (its index) differ in dimension')
+    len = size (X)
+    allocate (IX(len))
+    do i=1, len
+       IX(i)=X(i)
+       IA(i)=i
+    enddo
+    call MAPL_Sort(IX,IA)
+    _RETURN(_SUCCESS)
+
+  end subroutine sort_index
+
+
+  subroutine apply_order_index_R8 (X, IA, rc)
+    use MAPL_SortMod
+    real(ESMF_KIND_R8), intent(inout) :: X(:)
+    integer, intent(in) :: IA(:)            ! index
+    integer, optional, intent(out) :: rc
+
+    integer :: i, len
+    real(ESMF_KIND_R8), allocatable :: XX(:)
+
+    _ASSERT (size(X)==size(IA), 'X and IA (its index) differ in dimension')
+    len = size (X)
+    allocate (XX(len))
+    XX(:) = X(:)
+    do i=1, len
+       X(i) = XX(IA(i))
+    enddo
```

Comment:
Check me on this, but I think vanilla Fortran vector indices does this operation.   If it does, please eliminate the unused local variables.   And then consider whether it is clearer to have this procedure or just a comment with the inlined expression below ...
```suggestion
 x = x(IA)
```

Reply:
Great!  This works!  Thanks!